app/services/store_policy_resolver.py

```
@@ -0,0 +1,641 @@
+        Returns:
+            StorePolicyConfig: Immutable validated policy
+
+        Raises:
+            PolicyNotFound: If UUID does not resolve (expected after cleanup)
+            PolicyParseError: If payload fails schema validation
+            PolicyValidationError: If payload violates constraints
+        """
+        store_product = db.session.query(StoreProduct).filter_by(policy_uuid=policy_uuid).first()
+
+        if not store_product:
+            raise PolicyNotFound(f"Policy UUID {policy_uuid} not found (may have been deleted)")
```

Ruff (0.16.0)

[warning] Avoid specifying long messages outside the exception class

(TRY003)